### PR TITLE
when entryValue is nil, emit the null from kvstore api

### PR DESCRIPTION
**Pull request by xgess**
in the kvstore cli api, a non-existent or deleted entry previously came out as an empty string. this change pulls in a new change to the avdl compiler so we can have it come out as `null` instead of `""` which we believe is friendlier and more intuitive. 

### before
```sh
> keybase kvstore api -m '{"method": "get", "params": {"options": {"team": "team", "namespace": "whatever", "entryKey": "nonexistent"}}}'
{"result":{"teamName":"team","namespace":"whatever","entryKey":"nonexistent","entryValue":"","revision":0}}
```

### after
```sh
{"result":{"teamName":"team","namespace":"whatever","entryKey":"nonexistent","entryValue":null,"revision":0}}
```

#### related PRs
* depends on the avdl compiler https://github.com/keybase/node-avdl-compiler/pull/52
* blocks js bot library https://github.com/keybase/keybase-bot/pull/237
* blocks python bot library https://github.com/keybase/pykeybasebot/pull/55
* blocks go bot library https://github.com/keybase/go-keybase-chat-bot/pull/70
* server touch to the docs to match the change here https://github.com/keybase/keybase/pull/5457

### CODE REVIEW

**Comment by xgess** [comment]
@keybase/picnicsquad

File: protocol/avdl/keybase1/kvstore.avdl [hunk 1]
```diff
@@ -7,7 +7,7 @@ protocol kvstore {
     string teamName;
     string namespace;
     string entryKey;
-    string entryValue;
+    @nullSerializable(true) union { null, string } entryValue;
```
Comment (xgess) [inline protocol/avdl/keybase1/kvstore.avdl:10]: this is the change. i created this decorator in https://github.com/keybase/node-avdl-compiler/pull/52 and used it here and then everything else is downstream.

File: protocol/package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
   },
   "homepage": "https://github.com/keybase/protocol",
   "dependencies": {
-    "avdl-compiler": "1.4.8",
+    "avdl-compiler": "https://github.com/keybase/node-avdl-compiler.git#xgess/picnic-1133-allow-serializing-nulls",
```
Comment (xgess) [inline protocol/package.json:20]: will change this once https://github.com/keybase/node-avdl-compiler/pull/52 is merged

File: go/service/kvstore.go [hunk 3]
```diff
@@ -131,21 +131,22 @@ func (h *KVStoreHandler) getKVEntryLocked(ctx context.Context, arg keybase1.GetK
...
 	if err != nil {
 		err = fmt.Errorf("error comparing the entry from the server to what's in the local cache: %s", err)
 		mctx.Debug("%+v: %s", entryID, err)
 		return res, err
 	}
-	var cleartext string
-	if apiRes.Ciphertext != "" /* deleted or non-existent */ {
-		cleartext, err = h.Boxer.Unbox(mctx, entryID, apiRes.Revision, apiRes.Ciphertext, apiRes.TeamKeyGen, apiRes.FormatVersion, apiRes.WriterUID, apiRes.WriterEldestSeqno, apiRes.WriterDeviceID)
+	var entryValue *string
+	if apiRes.Ciphertext != nil && len(*apiRes.Ciphertext) > 0 /* deleted or non-existent */ {
```
Comment (patrickxb) [inline go/service/kvstore.go:141]: comment on this line is a bit confusing now...

**Pull request merged**